common/src/main/java/io/netty/util/concurrent/DefaultPromise.java

```diff
@@ -652,6 +652,8 @@ private void rethrowIfFailed() {
     }
 
     private boolean await0(long timeoutNanos, boolean interruptable) throws InterruptedException {
+        final long startTime = System.nanoTime();
```

Comment:
Getting the time can be expensive depending on the clock source. To avoid that cost when we don't need the start time, it should be moved back down to where it was.

Reply:
Isn't `System#nanoTime()` based on JNI?

Reply:
Depends on the JVM, OS, and sometimes on the hardware as well. It can be anything from a few instructions querying a TSC register, to a system call that involves hardware interrupts.

Reply:
From the logic perspective, `startTime` should count time from the beginning of the method.
From the performance perspective, if `System.nanoTime()` really hurt the performance, then I think it's OK to move it down.

Reply:
`System.nanoTime()` been moved down. @chrisvest